windows/shared/logging/PlogTraceLogging/include/PlogTraceLogging/PlogTraceLogging.hxx

```diff
@@ -0,0 +1,46 @@
+
+#ifndef PLOG_TRACE_LOGGING_HXX
+#define PLOG_TRACE_LOGGING_HXX
+
+#include <windows.h>
+
+#include <TraceLoggingProvider.h>
+
+#include <plog/Appenders/IAppender.h>
+#include <plog/Log.h>
+
+#include <string>
+
+TRACELOGGING_DECLARE_PROVIDER(g_hplogTraceLoggingProvider);
+
+namespace plog
+{
+
+template <class Formatter>
+class TraceLoggingAppender : public IAppender
+{
+public:
+    TraceLoggingAppender()
+    {
+        TraceLoggingRegister(g_hplogTraceLoggingProvider);
```

Comment:
The result of this should probably be checked. Per the [documentation](https://learn.microsoft.com/en-us/windows/win32/api/traceloggingprovider/nf-traceloggingprovider-traceloggingregister), you can use the `SUCCEEDED()` macro on the result.